R/fct_tables.R

```diff
@@ -51,12 +51,23 @@ create_table.default <- function(
   stopifnot(is.character(keep_vars))
   stopifnot(is.character(name_column))
   stopifnot(is.character(value_column))
+  if ("reviewed" %in% names(data)) {
+    data <- dplyr::mutate(
+      data,
+      o_reviewed = dplyr::case_when(
+        any(reviewed == "No") & any(reviewed == "Yes") ~ list(list(reviewed = NA, ids = id)),
+        any(reviewed == "Yes") ~ list(list(reviewed = TRUE, ids = id)),
+        .default = list(list(reviewed = FALSE, ids = id))
+      ),
+      .by = dplyr::all_of(keep_vars))
+    keep_vars <- c("o_reviewed", keep_vars)
+  }
```

Comment:
Still looking into the structure of everything, including the `o_reviewed` object. But I feel like this is an important (first?) step in the entire process, right? What is reviewed now, is that only `TRUE` if all items are reviewed?

Reply:
Correct, it is only true if all records contained in the row are marked as `reviewed`. Naming could be better. It's kind of an "overall reviewed" for the row.

Reply:
I wonder if we should extract this in a separate helper function since it is now such a pivotal step. That way, we can explain what is going on in a Roxygen tab, and we can write some simple tests for it. The o_reviewed object is now completely undocumented, it is hard to see what is needed in this object, what columns we expect here, and which would trigger an update etc.

Reply:
https://github.com/openpharma/clinsight/blob/f5fa833a2b5e9cd6caed609b27ede955d87b34b2/R/fct_tables.R#L71-L97

Reply:
https://github.com/openpharma/clinsight/blob/f5fa833a2b5e9cd6caed609b27ede955d87b34b2/R/fct_form_helpers.R#L62-L88